src/components/App/Page.js

```diff
@@ -18,26 +20,54 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import * as React from 'react';
 import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import type { Location } from 'react-router-dom';
 
 import TopNav from './TopNav';
+import { trackPageView } from '../../utils/metrics';
 
 import './Page.css';
 
-export default function JaegerUIPage({ children }) {
-  return (
-    <section className="jaeger-ui-page" id="jaeger-ui">
-      <Helmet title="Jaeger UI" />
-      <TopNav />
-      <div className="jaeger-ui--content">
-        {children}
-      </div>
-    </section>
-  );
+type PageProps = {
+  location: Location,
+  children: React.Node,
+};
+
+class Page extends React.Component<PageProps> {
+  props: PageProps;
+
+  componentDidMount() {
+    const { pathname, search } = this.props.location;
+    trackPageView(pathname, search);
```

Comment:
Nice, I think this is more robust. 

One comment, given that the GA key will come through the HTTP Config service (async), we need to make sure the key is initialized beforehand so we don't miss any page view events

Reply:
Yeah, I might make a separate ticket for that. Don't want to hold up fixing GA tracking or getting the menu working.